### PR TITLE
Refactor duplicate search code in homejs and devicejs

### PR DESCRIPTION
**Pull request by pkong-ds**

### CODE REVIEW

File: src/components/DeviceSearch/DeviceSearch.astro [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+<script define:vars={{ modelItems, brandItems, containerId: id }}>
+  window.modelItems = modelItems;
+  window.brandItems = brandItems;
+  if (window.containerIds != null) {
+    window.containerIds.push(containerId);
+  } else {
+    window.containerIds = [containerId];
+  }
+
+  console.log("hihi", containerId);
```
Comment (YayunHuang) [inline src/components/DeviceSearch/DeviceSearch.astro]: debug log? :thinking_face:

File: src/components/DeviceSearch/search.js [hunk 2]
```diff
@@ -167,6 +154,18 @@ function initializeAutocomplete(viewModel) {
...
     },
   });
+}
+
+function initialize() {
+  const { modelItems, brandItems, containerIds } = injectVariables();
+  console.log(modelItems, brandItems, containerIds);
+  containerIds.forEach((containerId) =>
+    console.log(document.querySelector(`#${containerId}`)),
+  );
```
Comment (YayunHuang) [inline src/components/DeviceSearch/search.js]: same

**Review by YayunHuang — CHANGES_REQUESTED** [review]
@pkong-ds please help to remove the log :joy:

**Comment by pkong-ds** [comment]
@YayunHuang sor, updated~

**Review by YayunHuang — APPROVED** [review]
LGTM, thanks

**Pull request merged**